server/models.py

```diff
@@ -782,8 +799,11 @@ def visible_grades(self):
         """ Return public grades. "Autograder" kind are errors from the
         autograder and should not be shown.
         """
+        published_scores = self.assignment.published_scores
+
         return [s for s in self.scores
-            if s.public and s.kind not in HIDDEN_GRADE_TAGS]
+            if s.public and s.kind not in HIDDEN_GRADE_TAGS and s.kind.title() in published_scores]
```

Comment:
This makes we wonder why `published_scores` are stored in title case. It's worth keeping it in lower case for consistency. 

It would also help make this list comprehension a bit shorted if it weren't in title case.